tests/backup/backup_restore_basic_test.go

```diff
@@ -1318,3 +1319,310 @@ var _ = Describe("{BackupSyncBasicTest}", func() {
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Unable to fetch px-central-admin ctx")
+			manualBkpSingleNs = fmt.Sprintf("%s-%v", "backup", time.Now().Unix())
+			err = CreateBackupWithNamespaceLabel(manualBkpSingleNs, SourceClusterName, backupLocationName, backupLocationUID,
+				labelSelectors, orgID, clusterUid, "", "", "", "", labelForSingleNamespace, ctx)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying status of backup [%s] creation with label [%s]", manualBkpSingleNs, labelForSingleNamespace))
+			err = NamespaceLabelBackupSuccessCheck(manualBkpSingleNs, ctx, []string{singleNamespace}, labelForSingleNamespace)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying if the labeled namespace [%v] is backed up and checks for labels [%s] applied to backup [%s]", singleNamespace, labelForSingleNamespace, manualBkpSingleNs))
+		})
+		Step("Taking manual backup of multiple applications with namespace label filter", func() {
+			log.InfoD("Taking manual backup of multiple applications with namespace label filter")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Unable to fetch px-central-admin ctx")
+			manualBkpMultipleNs = fmt.Sprintf("%s-%v", "multiple-namespace-backup", time.Now().Unix())
+			err = CreateBackupWithNamespaceLabel(manualBkpMultipleNs, SourceClusterName, backupLocationName, backupLocationUID,
+				labelSelectors, orgID, clusterUid, "", "", "", "", labelForMultipleNamespace, ctx)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying backup [%s] creation with labels [%v]", manualBkpMultipleNs, labelForMultipleNamespace))
+			err = NamespaceLabelBackupSuccessCheck(manualBkpMultipleNs, ctx, multipleNamespace, labelForMultipleNamespace)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying if the labeled namespaces [%v] are backed up and check for labels [%s] applied to backups [%s]", multipleNamespace, labelForMultipleNamespace, manualBkpMultipleNs))
+		})
+		Step("Taking manual backup of all applications with namespace label filter", func() {
+			log.InfoD("Taking manual backup of all applications with namespace label filter")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Unable to fetch px-central-admin ctx")
+			manualBkpAllNs = fmt.Sprintf("%s-%v", "all-namespace-backup", time.Now().Unix())
+			err = CreateBackupWithNamespaceLabel(manualBkpAllNs, SourceClusterName, backupLocationName, backupLocationUID,
+				labelSelectors, orgID, clusterUid, "", "", "", "", labelForAllNamespace, ctx)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying backup [%s] creation with labels [%v]", manualBkpAllNs, labelForAllNamespace))
+			err = NamespaceLabelBackupSuccessCheck(manualBkpAllNs, ctx, bkpNamespaces, labelForAllNamespace)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying if the labeled namespaces [%v] are backed up and check for labels [%s] applied to backups [%s]", bkpNamespaces, labelForAllNamespace, manualBkpAllNs))
+		})
```

Comment:
In all three scenarios most of the code is repeated only. You can write a common method where you can pass the labels and applied labels namespaces and method will return the backupname.